### PR TITLE
Merge develop into future

### PR DESCRIPTION
**Pull request by asbjornu**
Merge HEAD of `develop` into `future`.

### CODE REVIEW

**Comment by asbjornu** [comment]
@BeeWarloc: Since a test fails for reasons I don't understand, I created a pull request to merge `develop` into `future` instead of directly merging. Do you understand why this test fails?

```c$
Pomona.SystemTests.RootResourceTests.Get_Root_Returns_Dictionary_Containing_Resource_Links
Pomona.Common.Web.WebClientException`1[Critters.Client.IErrorStatus] : The GET request to <http://test/> failed with '500 InternalServerError': Pomona.UnitTests.UriResolverTests.BaseUri is null.
   at Pomona.Common.RequestDispatcher.SendRequest(String uri, String httpMethod, Object body, ISerializationContextProvider provider, RequestOptions options) in C:\projects\pomona\app\Pomona.Common\RequestDispatcher.cs:line 202
   at Pomona.Common.PomonaClient.Get(String uri, Type type, RequestOptions requestOptions) in C:\projects\pomona\app\Pomona.Common\PomonaClient.cs:line 165
   at Pomona.Common.RootResource`1.Get(String uri, Type type, RequestOptions requestOptions) in C:\projects\pomona\app\Pomona.Common\RootResource.cs:line 129
   at Pomona.Common.PomonaClientExtensions.Get[T](IPomonaClient client, String uri) in C:\projects\pomona\app\Pomona.Common\PomonaClientExtensions.cs:line 25
   at Pomona.SystemTests.RootResourceTests.Get_Root_Returns_Dictionary_Containing_Resource_Links() in C:\projects\pomona\tests\Pomona.SystemTests\RootResourceTests.cs:line 22
```

I have made attempts at ensuring that `Pomona.UnitTests.UriResolverTests.BaseUri` is not `null` when the test is run, but then I get into problems with it being the wrong string (the test expects `http://test/`, which I don't want to just mindlessly hard code somewhere else, as I don't exactly understand why the test is expecting that exact string).

Ideas?

File: app/Pomona/Routing/DataSourceRouteActionResolver.cs [hunk 1]
```diff
@@ -69,15 +71,12 @@ private IPomonaDataSource GetDataSource(IPomonaSession session)
             return pr =>
             {
                 var request = pr;
-                var uriResolver = request.Session.UriResolver;
```
Comment (BeeWarloc) [inline app/Pomona/Routing/DataSourceRouteActionResolver.cs]: @asbjornu You should have chosen the line above during conflict resolution, that's why the test is failing. 😺

**Comment by BeeWarloc** [comment]
Btw, I also see now that the ClientCompatibilityTests broke somewhere during some recent change to `develop`. I assume this slipped through as a result of `[Category("FailsOnFileShare")]` not working on your dev setup? :)

File: app/Pomona/Routing/DataSourceRouteActionResolver.cs (hunk 1 above)
Comment (asbjornu) [inline app/Pomona/Routing/DataSourceRouteActionResolver.cs]: Aha, thanks!

**Comment by asbjornu** [comment]
@BeeWarloc: Are the `ClientCompatibilityTests` broken? Both Travis and AppVeyor seem to be green to me:

[![Develop][1]][2] [![Develop][3]][4]

But yes, I added `[Category("FailsOnFileShare")]` to avoid running those tests when working on the code from a Windows file share. Are any of the tests I decorated with this attribute red anywhere now?

[1]: https://travis-ci.org/Pomona/Pomona.svg?branch=develop
[2]: https://travis-ci.org/Pomona/Pomona
[3]: https://img.shields.io/appveyor/ci/Pomona/Pomona/develop.svg
[4]: https://ci.appveyor.com/project/Pomona/Pomona/branch/develop

**Comment by BeeWarloc** [comment]
@asbjornu Yeah, one test is failing, not sure exactly why. But that's the case in `develop` too, so shouldn't be a blocker for mergeing this.

The tests don't seem to be run on AppVeyor, haven't checked Travis.

**Comment by asbjornu** [comment]
@BeeWarloc: Weird. Any idea why the tests aren't being run?

**Pull request merged**